### PR TITLE
Update to SDK support for new generation model.

### PR DESCRIPTION
**Pull request by chrimc62**
Add --dialog to specify dialog to run.

### CODE REVIEW

File: experimental/generation/runbot/Properties/launchSettings.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "profiles": {
     "IIS Express": {
       "commandName": "IISExpress",
-      "commandLineArgs": "--root \"C:\\tmp\\demo\\bot 6\"",
+      "commandLineArgs": "--root \"C:\\Users\\chrim\\source\\repos\\botbuilder-dotnet\\tests\\Microsoft.Bot.Builder.Dialogs.Adaptive.Tests\\Tests\\GeneratorTests\\sandwich\"",
```
Comment (GeoffCoxMSFT) [inline experimental/generation/runbot/Properties/launchSettings.json:13]: maybe this file shouldn't be checked in?

File: experimental/generation/runbot/RunBot.cs [hunk 2]
```diff
@@ -19,34 +19,37 @@
...
         {
-            this.dialogStateAccessor = conversationState.CreateProperty<DialogState>("RootDialogState");
-            this.resourceExplorer = resourceExplorer;
+            _dialogStateAccessor = conversationState.CreateProperty<DialogState>("RootDialogState");
+            _resourceExplorer = resourceExplorer;
+            _configuration = configuration;
 
             // auto reload dialogs when file changes
-            this.resourceExplorer.Changed += (sender, resources) =>
+            resourceExplorer.Changed += (sender, resources) =>
```
Comment (GeoffCoxMSFT) [inline experimental/generation/runbot/RunBot.cs:40]: Nit: _resourceExplorer

File: experimental/generation/runbot/RunBot.cs [hunk 3]
```diff
@@ -19,34 +19,37 @@
...
+using Microsoft.Extensions.Configuration;
 
 namespace RunBotServer
 {
     public class RunBot : ActivityHandler
     {
-        private IStatePropertyAccessor<DialogState> dialogStateAccessor;
-        private DialogManager dialogManager;
-        private readonly ResourceExplorer resourceExplorer;
+        private IStatePropertyAccessor<DialogState> _dialogStateAccessor;
```
Comment (GeoffCoxMSFT) [inline experimental/generation/runbot/RunBot.cs:28]: I'm a big fan of the underscore prefix as a style as it reduces the chance of bugs :)

File: experimental/generation/runbot/RunBot.cs [hunk 4]
```diff
@@ -73,14 +76,15 @@ private void LoadDialogs()
 
             Dialog lastDialog = null;
             var choices = new ChoiceSet();
+            var dialogName = _configuration["dialog"];
 
-            foreach (var resource in this.resourceExplorer.GetResources(".dialog").Where(r => r.Id.EndsWith(".main.dialog")))
+            foreach (var resource in _resourceExplorer.GetResources(".dialog").Where(r => dialogName != null ? r.Id == dialogName : true))
```
Comment (GeoffCoxMSFT) [inline experimental/generation/runbot/RunBot.cs:81]: consider case-insensitive string compare here.

File: experimental/generation/runbot/RunBot.cs (hunk 3 above)
Comment (chrimc62) [reply experimental/generation/runbot/RunBot.cs:28]: Me too.  I inherited this code.  You want me to change it all?

File: experimental/generation/runbot/RunBot.cs (hunk 4 above)
Comment (GeoffCoxMSFT) [inline experimental/generation/runbot/RunBot.cs:81]: Nit: There's a little extra work going on to compare dialogName != null for each r.
```
var dialogResources = _resourceExplorer.GetResources(".dialog");
dialogResources = dialogName != null ? dialogResources .Where(r => string.Equals(r.Id, dialogName, StringComparison.OrdinalIgnoreCase)) : dialogResources ;
foreach (var resource in dialogResources) {
```

**Pull request merged**